src/MagicSet.cpp

```diff
@@ -1122,17 +1122,15 @@ bool MagicSetTransformer::transform(AstTranslationUnit& translationUnit) {
                 // also - update input directives to correctly use default fact file names
                 if (ioTypes->isInput(originalRelation)) {
                     for (AstIO* io : program->getIOs()) {
-                        if (io->getQualifiedName() != originalName || io->getKVP("operation") != "input") {
+                        if (io->getQualifiedName() != originalName && io->getType() != AstIO::InputIO) {
```

Comment:
Are you sure about this? Won't this change the name for unrelated IOs now?

Reply:
> The problem is formatting and evaluation/magic_dfa not producing the expected csv output.

Oh dear...